src/snyk/common/commands/commandController.ts

```diff
@@ -0,0 +1,106 @@
+import { IOpenerService } from '../services/openerService';
+import { OpenIssueCommandArg, OpenCommandIssueType } from './types';
+
+export class CommandController {
+  private debouncedCommands: Record<string, _.DebouncedFunc<(...args: unknown[]) => Promise<unknown>>> = {};
+
+  constructor(
+    private openerService: IOpenerService,
+    private snykCode: ISnykCodeService,
+    private ossService: OssService,
+  ) {}
+
+  openBrowser(url: string): unknown {
+    return this.executeCommand(SNYK_OPEN_BROWSER_COMMAND, this.openerService.openBrowserUrl.bind(this), url);
+  }
+
+  copyAuthLink(): unknown {
+    return this.executeCommand(SNYK_COPY_AUTH_LINK_COMMAND, this.openerService.copyOpenedUrl.bind(this.openerService));
+  }
+
+  openLocal(path: vscode.Uri, range?: vscode.Range): void {
+    // todo: add error reporting
+    void vscode.window.showTextDocument(path, { viewColumn: vscode.ViewColumn.One, selection: range });
+  }
+
+  openSettings(): void {
+    void vscode.commands.executeCommand(VSCODE_GO_TO_SETTINGS_COMMAND, `@ext:${SNYK_PUBLISHER}.${SNYK_NAME_EXTENSION}`);
+  }
+
+  async createDCIgnore(custom = false, path?: string): Promise<void> {
+    if (!path) {
+      const paths = (vscode.workspace.workspaceFolders || []).map(f => f.uri.fsPath);
+      const promises = [];
+      for (const p of paths) {
+        promises.push(createDCIgnore(p, custom));
+      }
+      await Promise.all(promises);
+    } else {
+      await createDCIgnore(path, custom);
+    }
+  }
+
+  async openIssueCommand(arg: OpenIssueCommandArg): Promise<void> {
+    if (arg.issueType == OpenCommandIssueType.CodeIssue) {
+      const issue = arg.issue as CodeIssueCommandArg;
+      const suggestion = this.snykCode.analyzer.findSuggestion(issue.message);
+      if (!suggestion) return;
+      // Set openUri = null to avoid opening the file (e.g. in the ActionProvider)
+      if (issue.openUri !== null)
+        await vscode.commands.executeCommand(
+          SNYK_OPEN_LOCAL_COMMAND,
+          issue.openUri || issue.uri,
+          issue.openRange || issue.range,
+        );
+      this.snykCode.suggestionProvider.show(suggestion.id, issue.uri, issue.range);
+      suggestion.id = decodeURIComponent(suggestion.id);
+
+      analytics.logIssueIsViewed({
+        ide: IDE_NAME,
+        issueId: suggestion.id,
+        issueType: suggestion.isSecurityType ? 'Code Security Vulnerability' : 'Code Quality Issue',
+        severity: severityAsText(suggestion.severity),
+      });
+    } else if (arg.issueType == OpenCommandIssueType.OssVulnerability) {
+      const issue = arg.issue as OssIssueCommandArg;
+      void this.ossService.showSuggestionProvider(issue);
+    }
+  }
+
+  async executeCommand(
+    name: string,
+    fn: (...args: unknown[]) => Promise<unknown>,
+    ...args: unknown[]
+  ): Promise<unknown> {
+    if (!this.debouncedCommands[name])
+      this.debouncedCommands[name] = _.debounce(
```

Comment:
curious why we need to debounce these commands (and what is the interval currently set to?)

Reply:
The current interval is 200ms.

Debouncing is there because a user can issue multiple commands when working with extension functionality (changing settings often that would re-trigger the whole scan via command, or initiating login by pressing twice on the "Authenticate" button that would create several browser instances for him). 

The code around wasn't changed but I'll need to revisit it at some point because debouncing might not be needed for all of the items.